### PR TITLE
Remove restriction for output behavior properties in components

### PR DESCRIPTION
**Pull request by CarlosFigueiraMSFT**
Currently the binder forbids output behavior properties in components from being called from outside the component; this is a scenario that we want to enable, so we are removing this restriction. Notice that there is no way to create such property today, so this won't have any immediate impact.

### CODE REVIEW

File: src/libraries/Microsoft.PowerFx.Core/Binding/Binder.cs [hunk 1]
```diff
@@ -4362,12 +4362,6 @@ private static bool IsValidAccessToScopedProperty(IExternalControl lhsControl, I
                    (currentControl.IsComponentControl ||
                    (currentControl.TopParentOrSelf is IExternalControl { IsComponentControl: false })))
                 {
-                    // Behavior property is blocked from outside the component.
-                    if (isBehaviorOnly)
-                    {
-                        return false;
```
Comment (jack-work) [inline src/libraries/Microsoft.PowerFx.Core/Binding/Binder.cs:4368]: I this something we should (or even can) write a test for?

Comment (jack-work) [reply src/libraries/Microsoft.PowerFx.Core/Binding/Binder.cs:4368]: In this codebase, anyway

**Review by jack-work — APPROVED** [review]
Pretty straightforward, just wondering about tests.

File: src/libraries/Microsoft.PowerFx.Core/Binding/Binder.cs (hunk 1 above)
Comment (CarlosFigueiraMSFT) [reply src/libraries/Microsoft.PowerFx.Core/Binding/Binder.cs:4368]: Not at this point.

**Review by Shpakh-MSFT — DISMISSED** [review]
:shipit:

File: src/libraries/Microsoft.PowerFx.Core/Binding/Binder.cs [hunk 2]
```diff
@@ -4361,12 +4361,6 @@ private static bool IsValidAccessToScopedProperty(IExternalControl lhsControl, I
                    (currentControl.IsComponentControl ||
```
Comment (lesaltzm) [inline src/libraries/Microsoft.PowerFx.Core/Binding/Binder.cs:4361]: the `isBehaviorOnly` argument to this function is now unused, consider removing it?

**Review by lesaltzm — APPROVED** [review]

File: src/libraries/Microsoft.PowerFx.Core/Binding/Binder.cs (hunk 2 above)
Comment (CarlosFigueiraMSFT) [reply src/libraries/Microsoft.PowerFx.Core/Binding/Binder.cs:4395]: Good idea, done

**Pull request merged**